src/dyn/dyn_ffi.cc

```diff
@@ -0,0 +1,267 @@
+// Copyright 2016-2019 Envoy Project Authors
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "include/proxy-wasm/wasm_vm.h"
+#include "include/proxy-wasm/wasm_api_impl.h"
+
+inline uint32_t wasmResultToWord(proxy_wasm::WasmResult result) {
+  return static_cast<size_t>(result);
+}
```

Comment:
There is some weird casting happening here:
1) The whole codebase and integrations are using `Word` type to abstract Wasm32 (32-bit) / NullVM (64-bit) native word size, but here you're removing the `Word` type and instead are casting it directly to `size_t` (64-bit) only to later truncate it to `uint32_t` (32-bit).
2) The function is named `wasmResultToWord`, but it does not return the `Word` type.
3) It seems that the only reason for this file is to get rid of `Word` and `WasmResult` types. Why?
4) Also, do the existing Proxy-Wasm plugins work out-of-the-box or does this create a new ABI surface?

Reply:
Yeah, I don't know what was running through my mind with this, perhaps I was worried about 32-bit WASM sizes not translating to 64-bit-possible ABI but that doesn't seem to be an issue -- and it's a big nop anyways.